### PR TITLE
Fixed saving `last_touched` date of `named_users` to STATE

### PR DESCRIPTION
**Pull request by avityk**
# Description:
The whole `named_users` table is reloaded on every data extraction

# The cause of the problem:
There are no `date_keys` for this entity, and none of date fields are specified in schema. But they are present in the row, so we can use them.

# Solution:
* Added `date_keys` to calling `named_users` entity sync
* As `date_keys` are not in the schema for this table, moved date comparison before the row transformation

### CODE REVIEW

File: tap_urban_airship/__init__.py
Comment (timvisher) [inline tap_urban_airship/__init__.py:135]: It appears that these fields are [undocumented](https://docs.urbanairship.com/api/ua/#schemas/nameduserresponsebody). If that's the case then we at a minimum need to add a comment here to that effect. I think it would also be prudent to add a `raise` in the event that our expectation that these fields are always found is violated. I think that would be handled by #7 so we should get that merged first.

**Review by timvisher — CHANGES_REQUESTED** [review]
Let's add some documentation.

File: tap_urban_airship/__init__.py
Comment (avityk) [reply tap_urban_airship/__init__.py:135]: This is what is being returned from Urban Airship API here:
```
{
  "named_user_id": "...",
  "tags": {...},
  "created": "...",
  "last_modified": "...",
  "channels": [
    {...}
  ]
}
```
So you are right, it doesn't match the API documentation. I will add a comment in the code.
And yes, the code that handles the situation when date fields are not present is in PR #7.

Comment (avityk) [reply tap_urban_airship/__init__.py:135]: @timvisher Added a comment, please check it out.

**Comment by luandy64** [comment]
I think we should hold off on reviewing this PR until #7 is taken care of.

**Comment by avityk** [comment]
@luandy64 Rebased to `master` after merging #7 and squashed to one commit.

**Comment by luandy64** [comment]
Can we get an updated gist of an example run verifying that things are looking good? After that I think we are good to merge this.

**Comment by avityk** [comment]
@luandy64 Yes, sure.
My `config.json`:
```
{
  "app_key": "...",
  "app_secret": "...",
  "start_date": "2017-01-17T20:32:05Z"
}
```
Snippet of logs:
```
{"stream": "named_users", "schema": {"type": "object", "properties": {"named_user_id": {"type": "string"}, "tags": {"type": "array", "items": {"type": ["string", "null"]}}, "channels": {"type": "array", "items": {"type": ["string", "null"]}}}}, "key_properties": ["named_user_id"], "type": "SCHEMA"}
  INFO GET https://go.urbanairship.com/api/named_users
{"stream": "named_users", "record": {"named_user_id": "...", "tags": [], "channels": ["..."]}, "type": "RECORD"}
...
{"stream": "named_users", "record": {"named_user_id": "...", "tags": [], "channels": ["...", "..."]}, "type": "RECORD"}
{"value": {"lists": "2018-11-12T17:05:38", "channels": "2018-11-20T19:08:54", "segments": "2017-01-17T20:32:05Z", "named_users": "2018-11-16T14:23:38"}, "type": "STATE"}
  INFO Sync completed
```
So, as you can see, the date for `named_users` in `STATE` is newer than the `start_date` in `config.json`.

**Comment by luandy64** [comment]
Thanks for working on this @avityk ! 👍

**Pull request merged**
